### PR TITLE
Prometheus: Specific code mode view that has no options, instead sharing options with builder

### PR DESCRIPTION
**Pull request by torkelo**
This PR updates the code mode for the new builder to use the new PromQueryBuilderOptions component

* Adds resolution option (when it's not 1) 
* Makes the Query type show Both in Explore. 

Before: 
Screenshot from 2022-02-10 18-52-33

After: (there is a margin-bottom that should not be there on the query field code box that makes the gap to the next EditorRow 2px to much can't fix that without breaking the editor when feature toggle is off) 
Screenshot from 2022-02-10 18-50-43
Screenshot from 2022-02-10 18-50-53

After we have removed the feature toggle and made this new editor the default we could move the Metrics browser to the top header row so it can be used from the builder mode as well. Will make the code editor UI cleaner. Somethign like this: 

Pretty clean right :smile: 
Screenshot from 2022-02-10 18-58-17

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilderOptions.tsx [hunk 1]
```diff
@@ -41,15 +41,19 @@ export const PromQueryBuilderOptions = React.memo<Props>(({ query, app, onChange
...
   const showExemplarSwitch = app !== CoreApp.UnifiedAlerting && !query.instant;
 
   return (
     <EditorRow>
       <QueryOptionGroup title="Options" collapsedInfo={getCollapsedInfo(query, formatOption)}>
         <EditorField
           label="Legend"
-          tooltip="Controls the name of the time series, using name or pattern. For example
-        {{hostname}} will be replaced with label value for the label hostname."
+          tooltip="Serie name override or template. Ex. {{hostname}} will be replaced with label value for hostname."
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilderOptions.tsx:56]:
```suggestion
          tooltip="Series name override or template. Ex. {{hostname}} will be replaced with label value for hostname."
```

**Review by ivanahuckova — COMMENTED** [review]
I like this a lot! Couple things/bugs not related to this PR - so up to you if we want to tackle them here or in separate PR.

1. When we select `Both` in Explore, the Exemplar toggle should be visible.
image

2. If we toggle Exemplar = true and switch to Instant, it should change Exemplar to false otherwise we get this UI:
image

3. Exemplars defaults to false. 
I personally think this is a good move to default to false, but I know that when the feature was implemented, there was a big push to defaulting to `true` for the discoverability of the feature. So I am wondering if this should default to true and if we want to change default behaviour, have specific PR for it.

**Comment by torkelo** [comment]
@aocenas 
> Looks neat. Seems though that we lost some things, like exemplars and the prometheus UI link. Or do we want to move them somewhere else?

The exemplars toggle was a bug, fixed now. Asked in #prometheus channel and seems like that link is very very / never used. So maybe we can remove it? I would like to gauge demand for it. It's also broken on GrafanaCloud (as Prometheus UI is not available there), so if we add it back we should have some fallback for Grafana Cloud.

**Comment by torkelo** [comment]
@ivanahuckova fixed  all of those issues. I agree Exemplars should not be enabled by default, feels like only a small minority of queries/visualizations will want exemplars.

**Review by ivanahuckova — APPROVED** [review]
LGTM! This probably also resolves [Default exemplars to false](https://github.com/grafana/grafana/discussions/44264) discussion/feature request.

**Pull request merged**